lib/SILOptimizer/Differentiation/JVPCloner.cpp

```diff
@@ -826,6 +869,7 @@ class JVPCloner::Implementation final
     auto &tanValDest = getTangentBuffer(si->getParent(), si->getDest());
     diffBuilder.emitStoreValueOperation(loc, tanValSrc, tanValDest,
                                         si->getOwnershipQualifier());
+    tangentBufferInitializationInfo[tanValDest] = Initialized;
```

Comment:
When the store ownership qualifier is `Init`, I think you'll need to mark `tanValSrc` as uninitialized.

Reply:
Also, according to [`store`](https://github.com/apple/swift/blob/master/docs/SIL.rst#id88)'s reference:
> `store %0 to %1 : $*T`
> If `%1` already references a value that requires release or other cleanup, that value must be loaded before being stored over and cleaned up.

We may need to switch on the initializedness of `tanValDest`. if it's already initialized, we call `destroy_addr` on it before emitting `store`.

Reply:
Is that true that `Init` uninitializes src? That looks unexpected.

Reply:
Actually nvm, I mistook it for `copy_addr`. Since the source is an object (not an address), it needs no further handling.